remoting/src/main/java/org/apache/rocketmq/remoting/protocol/RemotingCommand.java

```diff
@@ -181,13 +181,12 @@ public static RemotingCommand decode(final ByteBuffer byteBuffer) throws Remotin
 
     public static RemotingCommand decode(final ByteBuf byteBuffer) throws RemotingCommandException {
         int length = byteBuffer.readableBytes();
-        int oriHeaderLen = byteBuffer.readInt();
-        int headerLength = getHeaderLength(oriHeaderLen);
+        int headerLength = byteBuffer.readInt();
```

Comment:
According to the frame layout, the first byte of the header length is used as a bit-field, flagging how the header is serialized.  Only the following 3 bytes are actually employed to represent header length.

Reply:
to optimize the code， support get protocol type from ByteBuf